core/src/mindustry/world/Tiles.java

```diff
@@ -27,64 +43,119 @@ public void each(Intc2 cons){
+     * @param x The location on the x-coordinate
+     * @param y The location on the y-coordinate
+     *
+     * @return a tile at coordinates, or null if out of bounds
+     */
     public @Nullable Tile get(int x, int y){
         return (x < 0 || x >= width || y < 0 || y >= height) ? null : array[y*width + x];
     }
 
-    /** @return a tile at coordinates; throws an exception if out of bounds */
+    /**
+     * Get a tile at coordinates with an exception
+     *
+     * @param x The location on the x-coordinate
+     * @param y The location on the y-coordinate
+     *
+     * @return a tile at coordinates
+     * @exception IllegalArgumentException throws this if out of bounds
+     */
     public @NonNull Tile getn(int x, int y){
         if(x < 0 || x >= width || y < 0 || y >= height) throw new IllegalArgumentException(x + ", " + y + " out of bounds: width=" + width + ", height=" + height);
         return array[y*width + x];
     }
 
-    /** @return a tile at coordinates, clamped. */
+    /**
+     * Get a tile at coordinates, and clamps the coordinates if they are out of bounds
+     *
+     * @param x The location on the x-coordinate
+     * @param y The location on the y-coordinate
+     *
+     * @return a tile at coordinates
+     */
     public @NonNull Tile getc(int x, int y){
         x = Mathf.clamp(x, 0, width - 1);
         y = Mathf.clamp(y, 0, height - 1);
         return array[y*width + x];
     }
-
-    /** @return a tile at an iteration index [0, width * height] */
+    /**
+     * Get a tile at an iteration index [0, width * height]
+     *
+     * @param idx The iteration index
+     *
+     * @return a tile at coordinates
+     */
     public @NonNull Tile geti(int idx){
         return array[idx];
     }
 
-    /** @return a tile at an int position (not equivalent to geti) */
+    /**
+     * Get a tile at an int position (not equivalent to geti)
+     *
+     * @param pos The int position (x and y coordinate packed into an integer)
+     *
+     * @return a tile at an int position
+     */
     public @Nullable Tile getp(int pos){
         return get(Point2.x(pos), Point2.y(pos));
     }
 
+    /**
+     * Loop through each tile
+     * @param cons The lambda to use to loop through each tile
+     */
     public void eachTile(Cons<Tile> cons){
         for(Tile tile : array){
             cons.get(tile);
         }
     }
 
+    /** @return The TileIterator tied to this tile array. */
     @Override
     public Iterator<Tile> iterator(){
-        //iterating through the entire map is expensive anyway, so a new allocation doesn't make much of a difference
+        // iterating through the entire map is expensive anyway, so a new allocation doesn't make much of a difference
         return new TileIterator();
     }
 
+    /**
+     * Implements through tiles with extra power
+     */
     private class TileIterator implements Iterator<Tile>{
+
+        /** The index at which the {@link TileIterator} lies. */
```

Comment:
this isn't public API and it's pretty obvious for anyone reading the source code. I think anyone can tell you that *"field index in class TileIterator"* is *"the index of the TileIterator"*